lib/Translation/EmitHLSCpp.cpp

```diff
@@ -1880,13 +1929,32 @@ void ModuleEmitter::emitModule(ModuleOp module) {
 #include <math.h>
 #include <stdint.h>
 #include <string.h>
-
-using namespace std;
-
 )XXX";
 
-  // Emit all functions in the call graph in a post order.
+  // Check for used Ip, emit corresponding include paths.
+  llvm::SmallDenseSet<hls::DeclareOp> emittedDeclareOp;
+  module->walk([&](Operation *op) {
+    if (auto instanceOp = dyn_cast<hls::InstanceOp>(op)) {
+      auto declaredOp = instanceOp.getDeclareOp();
+      if (!emittedDeclareOp.contains(declaredOp)) {
+        declaredOp.walk([&](Operation *nestedOp) {
+          if (nestedOp->getName().getStringRef() == "hls.uip.include") {
+            auto curPath = nestedOp->getAttr("paths").dyn_cast<ArrayAttr>()[0];
+            os << "#include ";
+            os << curPath;
+            os << "\n";
+            emittedDeclareOp.insert(declaredOp);
```

Comment:
You probably can create a `SmallDenseSet<Attribute> emittedIncludeAttrs` here, so that it ensures you never print duplicated includes.